plugins/store/guest-js/index.ts

```diff
@@ -8,50 +8,250 @@ import { invoke, Resource } from '@tauri-apps/api/core'
+
+  async get<T>(key: string): Promise<T | undefined> {
+    return (await this.store).get<T>(key)
+  }
+
+  async has(key: string): Promise<boolean> {
+    return (await this.store).has(key)
+  }
+
+  async delete(key: string): Promise<boolean> {
+    return (await this.store).delete(key)
+  }
+
+  async clear(): Promise<void> {
+    await (await this.store).clear()
+  }
+
+  async reset(): Promise<void> {
+    await (await this.store).reset()
+  }
+
+  async keys(): Promise<string[]> {
+    return (await this.store).keys()
+  }
+
+  async values<T>(): Promise<T[]> {
+    return (await this.store).values<T>()
+  }
+
+  async entries<T>(): Promise<Array<[key: string, value: T]>> {
+    return (await this.store).entries<T>()
+  }
+
+  async length(): Promise<number> {
+    return (await this.store).length()
+  }
+
+  async load(): Promise<void> {
+    await (await this.store).load()
+  }
+
+  async save(): Promise<void> {
+    await (await this.store).save()
+  }
+
+  async onKeyChange<T>(
+    key: string,
+    cb: (value: T | undefined) => void
+  ): Promise<UnlistenFn> {
+    return (await this.store).onKeyChange<T>(key, cb)
+  }
+
+  async onChange<T>(
+    cb: (key: string, value: T | undefined) => void
+  ): Promise<UnlistenFn> {
+    return (await this.store).onChange<T>(cb)
+  }
+
+  async close(): Promise<void> {
+    if (this._store) {
+      await (await this._store).close()
+    }
+  }
+}
+
+/**
+ * A key-value store persisted by the backend layer.
+ */
+export class Store extends Resource implements IStore {
+  private constructor(rid: number) {
     super(rid)
   }
 
   /**
-   * Inserts a key-value pair into the store.
+   * Create a new store.
    *
-   * @param key
-   * @param value
-   * @returns
+   * If the store already exists, its data will be overwritten.
+   *
+   * If the store is already loaded you must use {@link getStore} instead.
+   *
+   * @example
+   * ```typescript
+   * import { Store } from '@tauri-apps/api/store';
+   * const store = await Store.create('store.json');
+   * ```
+   *
+   * @param path Path to save the store in `app_data_dir`
+   * @param options Store configuration options
+   *
+   * @throws If a store at that path already exists
+   */
+  static async create(path: string, options?: StoreOptions): Promise<Store> {
+    const rid = await invoke<number>('plugin:store|create_store', {
+      path,
+      ...options
+    })
+    return new Store(rid)
+  }
+
+  /**
+   * Create a new Store or load the existing store with the path.
+   *
+   * If the store is already loaded you must use {@link getStore} instead.
+   *
+   * @example
+   * ```typescript
+   * import { Store } from '@tauri-apps/api/store';
+   * let store = await Store.get('store.json');
+   * if (!store) {
+   *   store = await Store.newOrExisting('store.json');
+   * }
+   * ```
+   *
+   * @param path Path to save the store in `app_data_dir`
+   * @param options Store configuration options
    */
+  static async newOrExisting(
```

Comment:
same as above, it should be `createNewOrLoadExisting` or `createNewOrGetExisting`